src/zope/index/text/tests/test_lexicon.py

```diff
@@ -244,15 +238,15 @@ def test_process_simple(self):
     def test_process_w_locale_awareness(self):
         import locale
         import sys
-        self._old_locale = locale.setlocale(locale.LC_ALL)
+        old_locale = locale.setlocale(locale.LC_ALL)
         # set German locale
         try:
-            if sys.platform == 'win32':
-                locale.setlocale(locale.LC_ALL, 'German_Germany.1252')
-            else:
-                locale.setlocale(locale.LC_ALL, 'de_DE.ISO8859-1')
-        except locale.Error:
+            locale_string = 'de_DE.ISO8859-1' if sys.platform != 'win32' else 'German_Germany.1252'
+            locale.setlocale(locale.LC_ALL, locale_string)
+        except locale.Error: # pragma: no cover
             return # This test doesn't work here :-(
```

Comment:
Might be a good idea to call `self.skipTest("de_DE.ISO8859-1 lcoale not available on this system")` now that we can rely on `skipTest()` existing, since we're on Python >= 2.7.